Bugfix: ComfoClime negativ temperature

Revilo91 opened the pull request:
I noticed that the outside temperature from the ComfoClime_api was unfortunately being received as a signed number. This meant that it was suddenly showing 6000°C outside. This pull request aims to fix that.

Revilo91 committed:
Bugfix: ComfoClime Negativ Temperature

msfuture commented:
Could you provide the output of your Dashboard JSON? As this isn't happening on my side.

Revilo91 commented:
<img width="404" height="293" alt="image" src="https://github.com/user-attachments/assets/7e620db2-3b84-494b-ae6d-6c2ca7afe08f" />

Here you can see at outdoorTemperature.. And I just noticed that it gets so hot (when it's charged like that, it's 6553.2°C).

I have no idea why that is...

Revilo91 commented:
> My supply pipe is buried about 30m outside in the garden to pre-warm in winter and pre-cool in summer. Even at -14°C peak this night near Ingolstadt, I had +4°C in the supply chain.

That explains why this has not yet been noticed. Unfortunately, my connection is not so protected. Near Regensburg it was -8°C and  the Comfoclime measured -7°C.

Revilo91 committed:
fix: replace _decode_raw_value with internal fix_temperature method in get_dashboard_data

Revilo91 committed:
Deleting unused constant

Revilo91 committed:
Refactor temperature handling and telemetry methods

- I've replaced the bitwise operations with Python-specific methods to make it more Pythonic.

- Additionally, I deleted a few lines that did not contribute to solving this problem.

- Revert: delete the property signed=True

Revilo91 committed:
Refactor byte conversion methods for clarity

after copilot review

Revilo91 committed:
Bugfix: Modify signed_int_to_bytes default signed parameter

Change default value of 'signed' parameter to False.

Revilo91 commented:
I think problems and doubts should now be resolved.

Revilo91 committed:
fix: Pass signed parameter to bytes_to_signed_int and signed_int_to_bytes methods
